app/services/hmpps_api/assessment_api.rb

```diff
@@ -0,0 +1,28 @@
+module HmppsApi
+  class AssessmentApi
+    def self.client
+      host = Rails.configuration.assessment_api_host
+      HmppsApi::Client.new(host)
+    end
+
+    def self.get_latest_oasys_date(offender_no)
+      safe_offender_no = URI.encode_www_form_component(offender_no)
+      route = "/offenders/nomisId/#{safe_offender_no}/assessments/summary?assessmentType=LAYER_3&assessmentStatus=COMPLETE"
+
+      return nil if get_all_assessments(route).empty? || get_all_assessments(route).is_a?(Hash) # returns a 404 status offender not found
+
+      Date.parse(returns_most_recent_assessment(get_all_assessments(route)))
```

Comment:
This could be refactored as a straightforward `if` statement.

Reply:
@ollietreend to check what the 404 Not Found response is – because it seems a bit odd right now!